src/PowerShellEditorServices/Language/TokenOperations.cs

```diff
@@ -21,100 +20,65 @@ internal static class TokenOperations
-            );
-
-            // Find matching comment regions   #region -> #endregion
-            foldableRegions.AddRange(
-                MatchCustomCommentRegionTokenElements(tokens, RegionKindRegion)
-            );
-
-            // Find blocks of line comments # comment1\n# comment2\n...
-            foldableRegions.AddRange(
-                MatchBlockCommentTokenElement(tokens, RegionKindComment)
-            );
-
-            // Find comments regions <# -> #>
-            foldableRegions.AddRange(
-                MatchTokenElement(tokens, TokenKind.Comment, RegionKindComment)
-            );
-
-            // Remove any null entries. Nulls appear if the folding reference is invalid
-            // or missing
-            foldableRegions.RemoveAll(item => item == null);
-
-            // Sort the FoldingReferences, starting at the top of the document,
-            // and ensure that, in the case of multiple ranges starting the same line,
-            // that the largest range (i.e. most number of lines spanned) is sorted
-            // first. This is needed to detect duplicate regions. The first in the list
-            // will be used and subsequent duplicates ignored.
-            foldableRegions.Sort();
-
-            // It's possible to have duplicate or overlapping ranges, that is, regions which have the same starting
-            // line number as the previous region. Therefore only emit ranges which have a different starting line
-            // than the previous range.
-            foldableRegions.RemoveAll( (FoldingReference item) => {
-                // Note - I'm not happy with searching here, but as the RemoveAll
-                // doesn't expose the index in the List, we need to calculate it. Fortunately the
-                // list is sorted at this point, so we can use BinarySearch.
-                int index = foldableRegions.BinarySearch(item);
-                if (index == 0) { return false; }
-                return (item.StartLine == foldableRegions[index - 1].StartLine);
-            });
+            for (int index = 0; index < tokens.Length; index++)
+            {
+                Token token = tokens[index];
+                if (token.Kind == TokenKind.Comment) {
+                    // Processing for comment regions <# -> #>
+                    if ((token.Kind == TokenKind.Comment) && (token.Extent.StartLineNumber != token.Extent.EndLineNumber)) {
+                        foldableRegions.Add(CreateFoldingReference(token, token, RegionKindComment));
+                    }
 
-            // Some editors have different folding UI, sometimes the lastline should be displayed
-            // If we do want to show the last line, just change the region to be one line less
-            if (ShowLastLine) {
-                foldableRegions.ForEach( item => { item.EndLine--; });
+                    if (IsBlockComment(index, tokens)) {
```

Comment:
Reading "block comment" I think the opposite of what this checks. Not sure if others agree

Reply:
Open to suggestions for a different name.

It's basically checking to see if the comment is on it's own line e.g.
`     # something`

as opposed to 

` $foo = 'bar' #something`